src/cpp/session/resources/grid/dtviewer.js

```diff
@@ -772,6 +772,16 @@ var invokeFilterPopup = function (ele, buildPopup, onDismiss, dismissOnClick) {
 };
 
 var createFilterUI = function(idx, col) {
+
+  // the index coming into this function is for absolute data purposes
+  // since this is a visual-centric function we operate based on the visible index
+  var visualIndex = idx -= columnOffset;
```

Comment:
Is it intentional to assign the result of `idx -= columnOffset` here? I'd personally prefer to see this as separate statements since it's difficult to read as-is.

Reply:
Nope! I would never do that intentionally, good catch. I originally was just using a shifted `idx` but added in the `visualIndex` later to be more clear and guess I missed the `-=`.